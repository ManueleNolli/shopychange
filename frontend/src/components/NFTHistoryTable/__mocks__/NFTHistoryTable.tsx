import { divWithoutChildrenMock } from '../../../../mocksUtils'

const NFTHistoryTableMock = () => divWithoutChildrenMock('NFTHistoryTable')
export default NFTHistoryTableMock
