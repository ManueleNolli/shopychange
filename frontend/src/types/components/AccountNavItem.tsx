import { ReactElement } from 'react'

export type AccountNavItem = {
  label: string
  icon?: ReactElement
  href?: string
  action?: () => void
}
