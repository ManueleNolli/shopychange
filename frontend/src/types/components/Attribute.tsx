export type Attribute = {
  traitType: string
  value: string
}
