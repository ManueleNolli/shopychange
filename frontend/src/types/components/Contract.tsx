export type Contract = {
  name: string
  address: string
  symbol: string
}
