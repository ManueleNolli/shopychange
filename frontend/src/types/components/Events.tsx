export enum EventType {
  Mint = 'EventMint',
  Burn = 'EventBurn',
  Transfer = 'EventTransfer',
  Approval = 'EventApproval',
  ApprovalForAll = 'EventApprovalForAll',
  SaleCreated = 'EventSaleCreated',
  SalePriceModified = 'EventSalePriceModified',
  SaleCancelled = 'EventSaleCancelled',
  SaleBought = 'EventSaleBought',
}

export interface EventGeneric {
  date: string
  __typename: EventType
}

export interface EventTransfer extends EventGeneric {
  fromAddress: string
  toAddress: string
}

export interface EventApproval extends EventGeneric {
  owner: string
  approved: string
}

export interface EventApprovalForAll extends EventGeneric {
  owner: string
  operator: string
  isApproved: boolean
}

export interface EventMint extends EventGeneric {
  toAddress: string
}

export interface EventBurn extends EventGeneric {
  fromAddress: string
}

export interface EventSaleCreated extends EventGeneric {
  seller: string
  price: number
}

export interface EventSalePriceModified extends EventGeneric {
  seller: string
  previousPrice: number
  price: number
}

export interface EventSaleCancelled extends EventGeneric {
  seller: string
}

export interface EventSaleBought extends EventGeneric {
  seller: string
  buyer: string
  price: number
}

export type EventNFT =
  | EventTransfer
  | EventApproval
  | EventApprovalForAll
  | EventMint
  | EventBurn
  | EventSaleCreated
  | EventSalePriceModified
  | EventSaleCancelled
  | EventSaleBought

export type NFTHistory = EventNFT[]
