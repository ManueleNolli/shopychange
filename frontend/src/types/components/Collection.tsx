import { Contract } from './Contract'
import { NFT } from './NFT'

export type Collection = {
  contract: Contract
  nfts: NFT[]
}
