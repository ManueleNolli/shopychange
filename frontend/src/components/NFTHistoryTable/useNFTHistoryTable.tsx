import { useState, useEffect } from 'react'
import { NFTHistory } from '../../types/components/Events'
import { NFTHistoryTableProps } from '../../types/props/NFTHistoryTableProps'
import { correctDate } from '../../utils/CorrectDate/CorrectDate'

export default function useNFTHistoryTable({
  nftHistory,
}: NFTHistoryTableProps) {
  const [history, setHistory] = useState<NFTHistory>([])

  useEffect(() => {
    // reverse order by timestamp (date like 1689700188) including seconds
    const copy = [...nftHistory]
    const reversedSortedData = copy.sort((a, b) => {
      return correctDate(b.date) - correctDate(a.date)
    })
    setHistory(reversedSortedData)
  }, [nftHistory])

  return {
    history,
  }
}
