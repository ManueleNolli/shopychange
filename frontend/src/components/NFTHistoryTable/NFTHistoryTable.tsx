import React from 'react'
import { NFTHistoryTableProps } from '../../types/props/NFTHistoryTableProps'
import {
  Center,
  Flex,
  Text,
  TableContainer,
  Table,
  Tbody,
  Td,
  Th,
  Thead,
  Tr,
} from '@chakra-ui/react'
import {
  EventApproval,
  EventApprovalForAll,
  EventMint,
  EventNFT,
  EventTransfer,
  EventType,
  EventBurn,
  EventSaleCreated,
  EventSalePriceModified,
  EventSaleCancelled,
  EventSaleBought,
} from '../../types/components/Events'
import { formatDate } from '../../utils/FormatDate/FormatDate'
import { correctDate } from '../../utils/CorrectDate/CorrectDate'

import { IoIosHammer } from 'react-icons/io' //Mint
import { BiTransfer } from 'react-icons/bi' //Transfer
import { IoCheckmarkDone } from 'react-icons/io5' //Approval
import { IoCheckmark } from 'react-icons/io5' //ApprovalForAll true
import { MdOutlineRemoveDone } from 'react-icons/md' //ApprovalForAll false
import { FaBurn } from 'react-icons/fa' //Burn
import { FaTimes } from 'react-icons/fa' //Cancel Sale
import { FaPen } from 'react-icons/fa' //Modify Sale
import { FaShoppingCart } from 'react-icons/fa' //Buy Sale
import { ImPriceTag } from 'react-icons/im' //Create Sale
import { FaLongArrowAltRight } from 'react-icons/fa'

import { renderAddress } from '../../utils/SimplifiedAddress/SimplifiedAddress'
import useNFTHistoryTable from './useNFTHistoryTable'

export default function NFTHistoryTable({ nftHistory }: NFTHistoryTableProps) {
  const { history } = useNFTHistoryTable({ nftHistory })

  return (
    <TableContainer maxW={{ base: '80vw', md: '70vw' }}>
      <Table variant="simple">
        <Thead>
          <Tr>
            <Th></Th>
            <Th>From / Seller / Owner</Th>
            <Th>To / Buyer / Approved</Th>
            <Th>Price</Th>
            <Th>Date</Th>
          </Tr>
        </Thead>
        <Tbody>
          {history.map((event, index) => {
            return renderEvent(event, index)
          })}
        </Tbody>
      </Table>
    </TableContainer>
  )
}

/* eslint-disable */
function renderEvent(event: EventNFT, index: number) {
  switch (event.__typename) {
    case EventType.Mint:
      return <MintEvent event={event as EventMint} key={index} />
    case EventType.Transfer:
      return <TransferEvent event={event as EventTransfer} key={index} />
    case EventType.Approval:
      return <ApprovalEvent event={event as EventApproval} key={index} />
    case EventType.ApprovalForAll:
      return (
        <ApprovalForAllEvent event={event as EventApprovalForAll} key={index} />
      )
    case EventType.Burn:
      return <BurnEvent event={event as EventBurn} key={index} />
    case EventType.SaleCreated:
      return <SaleCreatedEvent event={event as EventSaleCreated} key={index} />
    case EventType.SalePriceModified:
      return (
        <SalePriceModifiedEvent
          event={event as EventSalePriceModified}
          key={index}
        />
      )
    case EventType.SaleCancelled:
      return (
        <SaleCancelledEvent event={event as EventSaleCancelled} key={index} />
      )
    case EventType.SaleBought:
      return <SaleBoughtEvent event={event as EventSaleBought} key={index} />
  }
}
/* eslint-enable */

type MintEventProps = {
  event: EventMint
}
function MintEvent({ event }: MintEventProps) {
  return (
    <Tr>
      <Td>
        <Flex>
          <IoIosHammer size={'24'} />
          <Text ml="2">Mint</Text>
        </Flex>
      </Td>
      <Td />
      <Td>{renderAddress(event.toAddress, `/account/${event.toAddress}`)}</Td>
      <Td />
      <Td>{formatDate(correctDate(event.date))}</Td>
    </Tr>
  )
}

type TransferEventProps = {
  event: EventTransfer
}
function TransferEvent({ event }: TransferEventProps) {
  return (
    <Tr>
      <Td>
        <Flex>
          <BiTransfer size={'24'} />
          <Text ml="2">Transfer</Text>
        </Flex>
      </Td>
      <Td>
        {renderAddress(event.fromAddress, `/account/${event.fromAddress}`)}
      </Td>
      <Td>{renderAddress(event.toAddress, `/account/${event.toAddress}`)}</Td>
      <Td />
      <Td>{formatDate(correctDate(event.date))}</Td>
    </Tr>
  )
}

type ApprovalEventProps = {
  event: EventApproval
}

function ApprovalEvent({ event }: ApprovalEventProps) {
  return (
    <Tr>
      <Td>
        <Flex>
          <IoCheckmark size={'24'} />
          <Text ml="2">Approval</Text>
        </Flex>
      </Td>
      <Td>{renderAddress(event.owner, `/account/${event.owner}`)}</Td>
      <Td>{renderAddress(event.approved, `/account/${event.approved}`)}</Td>
      <Td />
      <Td>{formatDate(correctDate(event.date))}</Td>
    </Tr>
  )
}

type ApprovalForAllEventProps = {
  event: EventApprovalForAll
}
function ApprovalForAllEvent({ event }: ApprovalForAllEventProps) {
  return (
    <Tr>
      <Td>
        <Flex>
          {event.isApproved ? (
            <IoCheckmarkDone size={'24'} />
          ) : (
            <MdOutlineRemoveDone size={'24'} />
          )}
          {event.isApproved ? (
            <Text ml="2">Approval For All</Text>
          ) : (
            <Text ml="2">Removed Approval For All</Text>
          )}
        </Flex>
      </Td>
      <Td>{renderAddress(event.owner, `/account/${event.owner}`)}</Td>
      <Td>{renderAddress(event.operator, `/account/${event.operator}`)}</Td>
      <Td />
      <Td>{formatDate(correctDate(event.date))}</Td>
    </Tr>
  )
}

type BurnEventProps = {
  event: EventBurn
}
function BurnEvent({ event }: BurnEventProps) {
  return (
    <Tr>
      <Td>
        <Flex>
          <FaBurn size={'24'} />
          <Text ml="2">Burn</Text>
        </Flex>
      </Td>
      <Td>
        {renderAddress(event.fromAddress, `/account/${event.fromAddress}`)}
      </Td>
      <Td />
      <Td />
      <Td>{formatDate(correctDate(event.date))}</Td>
    </Tr>
  )
}

type SaleCreatedEventProps = {
  event: EventSaleCreated
}
function SaleCreatedEvent({ event }: SaleCreatedEventProps) {
  return (
    <Tr>
      <Td>
        <Flex>
          <ImPriceTag size={'18'} />
          <Text ml="2">Sale Created</Text>
        </Flex>
      </Td>
      <Td>{renderAddress(event.seller, `/account/${event.seller}`)}</Td>
      <Td />
      <Td>{event.price}</Td>
      <Td>{formatDate(correctDate(event.date))}</Td>
    </Tr>
  )
}

type SalePriceModifiedEventProps = {
  event: EventSalePriceModified
}
function SalePriceModifiedEvent({ event }: SalePriceModifiedEventProps) {
  return (
    <Tr>
      <Td>
        <Flex>
          <FaPen size={'18'} />
          <Text ml="2">Sale Modified</Text>
        </Flex>
      </Td>
      <Td>{renderAddress(event.seller, `/account/${event.seller}`)}</Td>
      <Td />
      <Td>
        <Flex>
          <Text>{event.previousPrice}</Text>
          <Center mx="2">
            <FaLongArrowAltRight size={'12'} />
          </Center>
          <Text>{event.price}</Text>
        </Flex>
      </Td>
      <Td>{formatDate(correctDate(event.date))}</Td>
    </Tr>
  )
}

type SaleCancelledEventProps = {
  event: EventSaleCancelled
}
function SaleCancelledEvent({ event }: SaleCancelledEventProps) {
  return (
    <Tr>
      <Td>
        <Flex>
          <FaTimes size={'24'} />
          <Text ml="2">Sale Cancelled</Text>
        </Flex>
      </Td>
      <Td>{renderAddress(event.seller, `/account/${event.seller}`)}</Td>
      <Td />
      <Td />
      <Td>{formatDate(correctDate(event.date))}</Td>
    </Tr>
  )
}

type SaleBoughtEventProps = {
  event: EventSaleBought
}
function SaleBoughtEvent({ event }: SaleBoughtEventProps) {
  return (
    <Tr>
      <Td>
        <Flex>
          <FaShoppingCart size={'18'} />
          <Text ml="2">Sale Bought</Text>
        </Flex>
      </Td>
      <Td>{renderAddress(event.seller, `/account/${event.seller}`)}</Td>
      <Td>{renderAddress(event.buyer, `/account/${event.buyer}`)}</Td>
      <Td>{event.price}</Td>
      <Td>{formatDate(correctDate(event.date))}</Td>
    </Tr>
  )
}
