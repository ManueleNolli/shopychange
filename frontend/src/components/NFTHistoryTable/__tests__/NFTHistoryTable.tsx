import React from 'react'
import { render } from '@testing-library/react'
import NFTHistoryTable from '../NFTHistoryTable'
import { EventType, NFTHistory } from '../../../types/components/Events'

jest.mock('../../../utils/SimplifiedAddress/SimplifiedAddress', () => ({
  renderAddress: jest.fn((address: string) => address),
}))

jest.mock('../../../utils/FormatDate/FormatDate', () => ({
  formatDate: jest.fn((data: string) => data),
}))

jest.mock('../../../utils/CorrectDate/CorrectDate', () => ({
  correctDate: jest.fn((data: string) => data),
}))

const mockProps: NFTHistory = [
  {
    __typename: EventType.Mint,
    date: '1688821020',
    toAddress: '0xEd6BE7e9d2A23373e1d2c87f17AE0780b814bced',
  },
  {
    __typename: EventType.Transfer,
    date: '1689775320',
    fromAddress: '0xEd6BE7e9d2A23373e1d2c87f17AE0780b814bced',
    toAddress: '0xe9eFbC61285d75198B3a58794E054C1F6aa44b25',
  },
  {
    __typename: EventType.Burn,
    date: '1689776028',
    fromAddress: '0xe9eFbC61285d75198B3a58794E054C1F6aa44b25',
  },
  {
    __typename: EventType.Approval,
    date: '1689775248',
    owner: '0xEd6BE7e9d2A23373e1d2c87f17AE0780b814bced',
    approved: '0x8121270572A0c7319f9854dC0f4e1e9D30d36E56',
  },
  {
    __typename: EventType.ApprovalForAll,
    date: '1689775668',
    owner: '0xe9eFbC61285d75198B3a58794E054C1F6aa44b25',
    isApproved: true,
    operator: '0x8121270572A0c7319f9854dC0f4e1e9D30d36E56',
  },
  {
    __typename: EventType.ApprovalForAll,
    date: '1689775812',
    owner: '0xe9eFbC61285d75198B3a58794E054C1F6aa44b25',
    isApproved: false,
    operator: '0x8121270572A0c7319f9854dC0f4e1e9D30d36E56',
  },
  {
    __typename: EventType.SaleCreated,
    date: '1689752112',
    seller: '0xe9eFbC61285d75198B3a58794E054C1F6aa44b25',
    price: 0.05,
  },
  {
    __typename: EventType.SaleBought,
    date: '1689752652',
    seller: '0xe9eFbC61285d75198B3a58794E054C1F6aa44b25',
    buyer: '0xEd6BE7e9d2A23373e1d2c87f17AE0780b814bced',
    price: 0.001,
  },
  {
    __typename: EventType.SaleCancelled,
    date: '1689752544',
    seller: '0xe9eFbC61285d75198B3a58794E054C1F6aa44b25',
  },
  {
    __typename: EventType.SalePriceModified,
    date: '1689752280',
    seller: '0xe9eFbC61285d75198B3a58794E054C1F6aa44b25',
    previousPrice: 0.05,
    price: 0.06,
  },
]

describe('NFTHistoryTable', () => {
  test('renders correctly', async () => {
    const { container } = render(<NFTHistoryTable nftHistory={mockProps} />)
    expect(container).toMatchSnapshot()
  })
})
