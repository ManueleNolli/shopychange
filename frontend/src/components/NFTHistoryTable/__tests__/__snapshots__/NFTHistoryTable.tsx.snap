// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NFTHistoryTable renders correctly 1`] = `
<div>
  <div
    class="chakra-table__container css-cf7hij"
  >
    <table
      class="chakra-table css-0"
    >
      <thead
        class="css-0"
      >
        <tr
          class="css-0"
        >
          <th
            class="css-0"
          />
          <th
            class="css-0"
          >
            From / Seller / Owner
          </th>
          <th
            class="css-0"
          >
            To / Buyer / Approved
          </th>
          <th
            class="css-0"
          >
            Price
          </th>
          <th
            class="css-0"
          >
            Date
          </th>
        </tr>
      </thead>
      <tbody
        class="css-0"
      >
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="css-k008qs"
            >
              <svg
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M277.6 246.9c-1.1-2-2.5-3.9-4.1-5.5l-18.4-18.2c-1.9-1.9-4.1-3.5-6.6-4.6-8.2-3.6-17.4-1.7-23.4 4.3-6.4 6.3-18.1 17.8-39.2 38.7-40.4 40-98.1 89.6-132 118.2-7.2 6.1-7.7 17-1.1 23.7L92 443.2c6.7 6.7 17.6 6.3 23.8-.9 29.4-34.6 79.5-92.3 119.4-131.8 20.5-20.2 32-31.8 38.5-38.2 6.5-6.6 8.3-16.7 3.9-25.4zM462.4 218.1l-34.5-34.2c-1.1-1.1-2.5-1.6-4-1.6-1.4 0-2.9.5-4 1.6-2.5 2.5-6.1 3.8-9.6 3.3-4.5-.5-9.3-1.9-12.4-4.9-7-6.9 1.1-20.5-5.1-29.2-6.2-8.7-16.1-19.8-23.2-26.9-7.1-7-35-33.7-82-52.9-16.6-6.8-32.8-9.3-47.3-9.3-26.5 0-47.4 8.5-54.8 15.3-5.5 5.1-11.2 14.1-3.1 14.1.7 0 1.5-.1 2.4-.2 4.5-.7 13.3-1.5 23.4-1.5 15.7 0 34.5 2.1 44.6 10.1 16.3 13.1 29.8 30.6 30.9 53.2.8 16.8-3.4 28.2-18.7 45.5-2.8 3.2-2.6 8 .4 10.9l19.2 19.2c3.1 3.1 8.1 3.1 11.2.1 14-13.6 22.1-20.2 31.3-22.7 4-1.1 8.4-1.6 12.8-1.6 9.7 0 19.2 2.2 23.6 4.6 1.1.6 2.1 1.4 3.1 2.4 6.5 6.6 6.1 17.4-.5 23.9l-2 1.9c-2.2 2.2-2.2 5.7 0 7.9l34.5 34.2c1.1 1.1 2.5 1.6 4 1.6 1.4 0 2.9-.5 4-1.6l55.8-55.2c2.1-2.3 2.1-5.8 0-8z"
                />
              </svg>
              <p
                class="chakra-text css-14gt3bc"
              >
                Mint
              </p>
            </div>
          </td>
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="css-k008qs"
            >
              <svg
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m15 12 5-4-5-4v2.999H2v2h13zm7 3H9v-3l-5 4 5 4v-3h13z"
                />
              </svg>
              <p
                class="chakra-text css-14gt3bc"
              >
                Transfer
              </p>
            </div>
          </td>
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="css-k008qs"
            >
              <svg
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 384 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M192 0C79.7 101.3 0 220.9 0 300.5 0 425 79 512 192 512s192-87 192-211.5c0-79.9-80.2-199.6-192-300.5zm0 448c-56.5 0-96-39-96-94.8 0-13.5 4.6-61.5 96-161.2 91.4 99.7 96 147.7 96 161.2 0 55.8-39.5 94.8-96 94.8z"
                />
              </svg>
              <p
                class="chakra-text css-14gt3bc"
              >
                Burn
              </p>
            </div>
          </td>
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="css-k008qs"
            >
              <svg
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 128L192 384l-96-96"
                  fill="none"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="32"
                />
              </svg>
              <p
                class="chakra-text css-14gt3bc"
              >
                Approval
              </p>
            </div>
          </td>
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="css-k008qs"
            >
              <svg
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M464 128L240 384l-96-96m0 96l-96-96m320-160L232 284"
                  fill="none"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="32"
                />
              </svg>
              <p
                class="chakra-text css-14gt3bc"
              >
                Approval For All
              </p>
            </div>
          </td>
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="css-k008qs"
            >
              <svg
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M4.84 1.98L3.43 3.39l10.38 10.38-1.41 1.41-4.24-4.24-1.41 1.41 5.66 5.66 2.83-2.83 6.6 6.6 1.41-1.41L4.84 1.98zm13.21 10.38L23 7.4 21.57 6l-4.94 4.94 1.42 1.42zm-.71-4.96l-1.41-1.41-2.12 2.12 1.41 1.41 2.12-2.12zM1.08 12.35l5.66 5.66 1.41-1.41-5.66-5.66-1.41 1.41z"
                />
              </svg>
              <p
                class="chakra-text css-14gt3bc"
              >
                Removed Approval For All
              </p>
            </div>
          </td>
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="css-k008qs"
            >
              <svg
                fill="currentColor"
                height="18"
                stroke="currentColor"
                stroke-width="0"
                version="1.1"
                viewBox="0 0 16 16"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.25 0h-6c-0.412 0-0.989 0.239-1.28 0.53l-7.439 7.439c-0.292 0.292-0.292 0.769 0 1.061l6.439 6.439c0.292 0.292 0.769 0.292 1.061 0l7.439-7.439c0.292-0.292 0.53-0.868 0.53-1.28v-6c0-0.412-0.338-0.75-0.75-0.75zM11.5 6c-0.828 0-1.5-0.672-1.5-1.5s0.672-1.5 1.5-1.5 1.5 0.672 1.5 1.5-0.672 1.5-1.5 1.5z"
                />
              </svg>
              <p
                class="chakra-text css-14gt3bc"
              >
                Sale Created
              </p>
            </div>
          </td>
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          >
            0.05
          </td>
          <td
            class="css-0"
          />
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="css-k008qs"
            >
              <svg
                fill="currentColor"
                height="18"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
                />
              </svg>
              <p
                class="chakra-text css-14gt3bc"
              >
                Sale Bought
              </p>
            </div>
          </td>
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          >
            0.001
          </td>
          <td
            class="css-0"
          />
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="css-k008qs"
            >
              <svg
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 352 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
              <p
                class="chakra-text css-14gt3bc"
              >
                Sale Cancelled
              </p>
            </div>
          </td>
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="css-k008qs"
            >
              <svg
                fill="currentColor"
                height="18"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
                />
              </svg>
              <p
                class="chakra-text css-14gt3bc"
              >
                Sale Modified
              </p>
            </div>
          </td>
          <td
            class="css-0"
          />
          <td
            class="css-0"
          />
          <td
            class="css-0"
          >
            <div
              class="css-k008qs"
            >
              <p
                class="chakra-text css-0"
              >
                0.05
              </p>
              <div
                class="css-1fkxuth"
              >
                <svg
                  fill="currentColor"
                  height="12"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
                  />
                </svg>
              </div>
              <p
                class="chakra-text css-0"
              >
                0.06
              </p>
            </div>
          </td>
          <td
            class="css-0"
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
